lib/modem_info/modem_info.c

```diff
@@ -503,7 +507,15 @@ static void modem_info_rsrp_subscribe_handler(void *context, const char *respons
 		return;
 	}
 
-	err = modem_info_parse(modem_data[MODEM_INFO_RSRP],
+	const struct modem_info_data rsrp_data = {
```

Comment:
How about rather than creating a local struct here, we add this as a new `static const struct modem_info_data rsrp_notify_data` below the already-existing `rsrp_data`? I think that might avoid some naming confusion.

Reply:
I considered that, but I did it locally specifically because it is not a requested value that is called by user. Only two members of the struct are used for parsing notification. The rest (`.cmd`, `.data_type`), etc. aren't used at all. 

It seemed cleaner to separate the two. 

I would have avoided struct all together and replaced `modem_info_parse` with local calls, but it would have meant code duplication. 

Since `modem_info_parse` only uses one member of the struct it can be changed to just accept `param_count` instead of the whole structure. This is irrespective of this particular usage. Then we wouldn't need a struct for this at all. We can just pass in the values, similar to how we call `at_params_short_get`.

In any event, I don't feel religious about my view so if you prefer your approach I can make that change.

Reply:
Fair points; the new struct can not be used the same way as the others. I would still rename it though, preferring not to have overlapping names if possible.

Reply:
IMO we could leave it as it is. Every other `modem_info_data` is linked with as corresponding `modem_info` enum making it accessible from the outside with appropriate API call. This one cannot be, so keeping it local seems fine.

As @bjda mentioned, we might need a more generic approach for handling notification, but it's outside of the scope of this PR.

Reply:
> I would still rename it though

Good point, @arvinf could you rename it to `rsrp_notify_data` as suggested?

Reply:
Done.